src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -104,8 +111,12 @@ private void OnMessage(Message msg)
                     OnPongMessageReceived((PingPayload)msg.Payload);
                     break;
                 case MessageCommand.Transaction:
+                    Transaction tx = (Transaction)msg.Payload;
+                    RenewKnownHashes(tx.Hash);
                     if (msg.Payload.Size <= Transaction.MaxTransactionSize)
```

Comment:
It seems that this check it's done twice in order to avoid the deserialization, if you cast first the payload it's useless